src/v/cloud_storage/remote_partition.cc

```diff
@@ -618,13 +661,20 @@ ss::future<> remote_partition::stop() {
           return segment->stop();
       });
 
-    // We may have some segment or reader objects enqueued for stop in
-    // the shared eviction queue: must flush it, or they can outlive
-    // us and trigger assertion in retry_chain_node destructor.
-    // This waits for all evictions, not just ours, but that's okay because
-    // stopping readers is fast, the queue is not usually long, and destroying
-    // partitions is relatively infrequent.
-    co_await materialized().flush_evicted();
+    // Do the last pass over the eviction list to stop remaining items returned
+    // from readers after the eviction loop stopped.
+    for (auto& rs : _eviction_pending) {
```

Comment:
instead of adding extra loop here you can handle `condition_variable_broken_exception` in the eviction loop and process remaining elements after that

Reply:
Done